internal/daemon/worker/proxy/options.go

```diff
@@ -51,3 +52,11 @@ func WithPostConnectionHook(fn func(net.Conn)) Option {
 		o.WithPostConnectionHook = fn
 	}
 }
+
+// WithDnsServer allows specifying lookup of the endpoint to happen via an
+// alternate DNS server. Must be in host:port form.
+func WithDnsServer(with string) Option {
```

Comment:
To me a `host:port` combination is an address, mind naming this as such to maybe help a bit?
```suggestion
func WithDnsServerAddress(with string) Option {
```

Reply:
Will make that change but do it in a separate commit so I update the comment too.